File: 박광열/14405.js
const fs = require("fs");
const input = fs.readFileSync("/dev/stdin").toString().trim().split("").reverse();
const dict = { pi: true, ka: true, chu: true };
let List = "";
while (input.length > 0) {
  List += input.pop();
  if (2 <= List.length && List.length < 4 && dict[List]) {
    List = "";
  } else if (List.length >= 4) {
    break;
  }
}
if (List == "" && input.length == 0) {
  console.log("YES");
} else {
  console.log("NO");
}

File: 김영아/14405.java
package 김영아;

public class 14405 {
    
    public static void main(String[] args){

        boolean yn=false;
        Scanner scanner=new Scanner(System.in);
        

        String input=scanner.nextLine();
        
        for (int i = 0; i < input.length(); ) {
            if (i + 1 < input.length() && input.substring(i, i + 2).equals("pi")) {
                i += 2;
            } else if (i + 1 < input.length() && input.substring(i, i + 2).equals("ka")) {
                i += 2;
            } else if (i + 2 < input.length() && input.substring(i, i + 3).equals("chu")) {
                i += 3;
            } else {
                yn = false;
                break;
            }
        }
        if (yn) {
            System.out.println("YES");
        } else {
            System.out.println("NO");
        }

        
    }
}
